warehouse/query-core/src/main/java/datawave/query/planner/DefaultQueryPlanner.java

```diff
@@ -1422,6 +1467,11 @@ protected ASTJexlScript processTree(final ASTJexlScript originalQueryTree, Shard
         return queryTree;
     }
     
+    private void recordPlan(ASTJexlScript queryTree, ShardQueryConfiguration config) {
+        this.plannedScript = JexlStringBuildingVisitor.buildQuery(queryTree);
```

Comment:
Is there an understanding of how expensive it might be to call buildQuery multiple times in the query planning process? As we're calling `recordPlan` many places throughout the code, and I'm assuming we ignore the result of the previous `buildQuery` call each time `recordPlan` is updated, we might consider a strategy where we call `buildQuery` in exception handling code only when it's needed? If `buildQuery` is cheap, no worries.

Reply:
I like the idea of only recording the query when the exception is thrown. Doing so will be far easier once https://github.com/NationalSecurityAgency/datawave/pull/1274 is merged in.

Reply:
@drewfarris and @apmoriarty, I agree with that concern and that's why I originally implemented the bugfix to avoid making so many calls to buildQuery.  All three of yesterday's reviewers disagreed so I changed the implementation.  

@ivakegg, @jwomeara and @hlgp do you happen to have an understanding of the relative expense of calling buildQuery after each step of the planning?  If not, perhaps I could measure it to better inform everyone.

Reply:
buildQuery is already being called multiple times throughout the planning process.  The cost of calling that method will vary depending on the number of nodes in the tree (which will vary from query to query), but I think it's generally a fast operation.

Here's a thought...  Why not add a queryTree variable to ShardQueryConfiguration instead of having a local variable within the DefaultQueryPlanner:process method.  Update the variable in ShardQueryConfiguration after each phase of the  planning process, and then if an exception gets thrown anywhere in the process, you can build the query when the exception is caught in RunningQuery using the last updated queryTree in ShardQueryConfiguration.  Everybody wins!

Reply:
That seems reasonable to me

Reply:
Done.

Note that RunningQuery is in datawave-ws-query which does not depend on the datawave-query-core so doesn't have access to JexlStringBuildingVisitor.  As such, I put the call to JexlStringBuildingVisitor.buildQuery in ShardQueryConfiguration.  I'm open to suggestions on this approach.